No decision based on `peer_id` in maker rollover decision making

da-kami opened the pull request:
The maker cannot base his decision on the CFD's peer_id being set, because previously opened positions will not contain the peer_id and the maker cannot default it.
The idea was that the maker would use the `peer_id` known in the connection to decide, but accept/reject are completely independent of the connection.

To keep this logic simple and avoid passing on state to the cfd actor we just try to dispatch to libp2p and if that fails (because there is no pending rollover) we fallback to the legacy connection.
This removes the need of load the CFD which is nice.

klochowicz reviewed (commented):
One drawback of this approach is that if for whatever reason accepting/rejecting won't work with a new taker (that uses libp2p), we'll try to use the old protocol with him, leading to more errors, as it cannot handle it. :/  (fortunately, it will only produce errors on the taker side, not a panic!)

I'm tempted to keep the `can_use_libp2p()` so that we know when we should *not* fallback (if we have `PeerId` stored).

The other alternative worth exploring would be keeping the old rollovers roll over using old protocol, that's perfectly reasonable too, as we will be supporting it for the foreseeable future.

klochowicz commented on daemon/src/libp2p_utils.rs:
```diff
@@ -48,11 +48,3 @@ pub fn create_listen_tcp_multiaddr(ip: &IpAddr, port: u16) -> Result<Multiaddr>
 }
-
-/// Determine whether to use libp2p or fallback to a legacy protocol
-pub fn can_use_libp2p(cfd: &model::Cfd) -> bool {
```
I'd prefer if you kept the function and used it to do the early exit

da-kami committed:
No decision based on `peer_id` in maker rollover decision making

The maker cannot base his decision on the CFD's peer_id being set, because previously opened positions will not contain the peer_id and the maker cannot default it.
The idea was that the maker would use the `peer_id` known in the connection to decide, but accept/reject are completely independent of the connection.

To keep this logic simple and avoid passing on state to the cfd actor we just try to dispatch to libp2p and if that fails (because there is no pending rollover) we fallback to the legacy connection.
This removes the need of load the CFD which is nice.

klochowicz commented on maker/src/cfd.rs:
```diff
@@ -529,31 +529,38 @@ impl<O, T, W> Actor<O, T, W> {
+                tracing::debug!("Try fallback to legacy rollover because unable to handle accept via libp2p: {error:#}");
+            }
+            Err(error) => {
+                // we should never see this given that the libp2p rollover actor is always running
+                tracing::error!("Try fallback to legacy rollover because unable to dispatch accept to libp2p rollover actor: {error:#}");
```
I would just return an error here. We should not be trying to use a different protocol because maker has errors; this will just incur errors on the taker because of maker's misbehavior.

da-kami replied:
Given that it might be a legacy scenario we have to fallback here.
We cannot just not rollover legacy requests because libp2p is broken.

But with the additional check for `peer_id` this should be fine.

klochowicz reviewed (approved):
I guess it's good for now. It will simplify itself when we get rid of the legacy connection anyway...

bonomat reviewed (commented):
Could we fix this differently if we auto-accept the rollover request if a request comes in via libp2p and keep the old mode of accepting using the legacy connection?
